File: apps/api/app/schemas/notification.py
from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel


class NotificationRead(BaseModel):
    id: UUID
    company_id: UUID
    appointment_id: UUID
    kind: str
    channel: str
    target: str | None = None
    payload: dict[str, Any]
    status: str
    delivered: bool
    delivered_at: datetime | None = None
    read_at: datetime | None = None
    created_at: datetime

    class Config:
        from_attributes = True

File: apps/mobile/src/types/notification.ts
export interface NotificationPayload {
  [key: string]: any;
}

export interface Notification {
  id: string;
  company_id: string;
  appointment_id?: string | null;
  kind: string;
  channel: string;
  target?: string | null;
  payload: NotificationPayload;
  status: string;
  delivered: boolean;
  delivered_at?: string | null;
  read_at?: string | null;
  created_at: string;
}
